### PR TITLE
feature: The Keymaster

### PR DESCRIPTION
**Pull request by yourbuddyconner**
Keymaster from Ghostbusters

### CODE REVIEW

File: tools/keymaster/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+## Using The Keymaster 
+
+Note: Before you do *anything*, [call the Ghostbusters](https://www.youtube.com/watch?v=Fe93CLbHjxQ). 
+
+The Keymaster is a simple Python-based CLI program, the entrypoint is `keymaster.py`
+
+Install the requirements via pip: 
+
+`pip3 install -r requirements.txt`
```
Comment (prestwich) [inline tools/keymaster/README.md:26]: how do you feel about pipenv

File: tools/keymaster/utils.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+from web3 import Web3
+
+# Checks if an address is below the threshold
+# returns difference in wei if true
+# returns False if not
+def is_wallet_below_threshold(address:str, threshold:int, endpoint:str):
```
Comment (prestwich) [inline tools/keymaster/utils.py:6]: need a linter. `: `

**Review by prestwich — COMMENTED** [review]
add mypy and pipenv

File: tools/keymaster/README.md (hunk 1 above)
Comment (yourbuddyconner) [reply tools/keymaster/README.md:26]: It is great, though I usually use `virtualenv`. I will add docs around it, just didn't get around to it after the first pass.

File: tools/keymaster/utils.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+# creates a transaction for a sender and recipient
+# given a network RPC endpoint
+# returns tuple (tx_params, signed_tx) for debugging
+def create_transaction(sender_key:str, recipient_address:int, amount:int, nonce:int, endpoint:str):
+    # Set up w3 provider with network endpoint
+    w3 = Web3(Web3.HTTPProvider(endpoint))
+    recipient_address = Web3.toChecksumAddress(recipient_address)
+    chain_id = w3.eth.chain_id
+    gas = 100000 * 100 if "arb-rinkeby" in endpoint else 100000
```
Comment (yourbuddyconner) [inline tools/keymaster/utils.py:38]: https://twitter.com/YourBuddyConner/status/143537217384614297/

**Review by prestwich — APPROVED** [review]

**Comment by prestwich** [comment]
approving with the assumption that we can clean up linting and type stuff later

**Pull request merged**